### ISSUE TITLE
feature request: cloudwatch_logs templating failure can lead to a very large number of warn logs

### ISSUE THREAD

**Issue by PettitWesley** [aws/aws-for-fluent-bit#599]
See: https://github.com/fluent/fluent-bit/issues/6918

The warning for record_accessor failure here can be emitted for every single log record processed: https://github.com/fluent/fluent-bit/blob/master/plugins/out_cloudwatch_logs/cloudwatch_api.c#L1084

It'd be ideal if this was only emitted once per batch instead. This will require code changes.

**Comment by PettitWesley** [comment]
For 2.0 there are workarounds: https://github.com/fluent/fluent-bit/issues/6918#issuecomment-1478346368

**Comment by PettitWesley** [comment]
This is a concern for the upcoming Daemon support launch, since it makes extensive use of templating which can fail .

**Comment by Mattie112** [comment]
Yes we also had this issue costing quite a bit (and yes we now have alerts). Would be great to have a solution indeed!

**Comment by petemounce** [comment]
It's not only templating that can fail with this impact, but also access denied when attempting to (say) create a log stream (where fluent bit _can_ create the log stream for its own logs for whatever misconfiguration reason).

**Comment by PettitWesley** [comment]
The new log suppress feature in 2.0 is good for the CW access denied and other errors. Which suppresses the same message from a single plugin. 

For the templating failures, the message actually comes from a core library which I think is not part of the log suppress feature.

**Comment by lexsca** [comment]
we ran into this as well and racked up massive cloudwatch changes. the [default values](https://github.com/aws/eks-charts/blob/463b7a4c386950a8f403e9d4129a9136e7973baa/stable/aws-for-fluent-bit/values.yaml#L122) should not have any template variables that aren't guaranteed to be there. people will probably keep tripping over this. 😐

**Comment by PettitWesley** [comment]
When Fluent Bit is deployed as a kubernetes daemonset pod, it collects its own logs, and thus, when it emits an error message, it will collect its own error message, potentially leading to a cycle of log spam in which each of its own error logs cause it to produce another error log. 

Key cases:
1. This issue- templating failure.
2. Log stream creation failure or API failure in general: if FLB can not create log streams or uploads, then this will lead to error logs, and then more error logs for the failure to upload those error logs.

**Comment by PettitWesley** [comment]
See one option for workaround here: https://github.com/fluent/fluent-bit/issues/6918#issuecomment-1811495825
